### PR TITLE
Merge dev/release/2.0.0 into master

### PR DESCRIPTION
**Pull request by eerhardt**
Merging build infrastructure changes (and other changes that went into release/2.0.0) into master.

### CODE REVIEW

**Comment by eerhardt** [comment]
NOTE: the Tizen and arm builds are expected to fail since I need change to netci.groovy.

**Comment by eerhardt** [comment]
Anyone have any feedback?  I plan on merging this at noon today, if none.

File: init-tools.sh [hunk 1]
```diff
@@ -12,126 +12,116 @@ __BUILD_TOOLS_PACKAGE_VERSION=$(cat $__scriptpath/BuildToolsVersion.txt)
...
-        *)
-            echo "Unsupported OS '$OSName' detected. Downloading linux-x64 tools."
-            OS=Linux
-            __DOTNET_PKG=dotnet-dev-linux-x64
-            ;;
-  esac
+
+    echo "Done initializing tools."
+    mkdir -p "$(dirname "$__BUILD_TOOLS_SEMAPHORE")" && touch $__BUILD_TOOLS_SEMAPHORE
+    exit 0
```
Comment (weshaggard) [inline init-tools.sh]: You can do this as a follow-up but I've sense changed this to be a return 0 as well. It is really only interesting in the source-build scenario though.

**Comment by eerhardt** [comment]
Looks like the last one passed.  https://devdiv.visualstudio.com/DevDiv/_build/index?context=allDefinitions&path=%5C&definitionId=3160&_a=completed

The one before that failed with:

```
2017-10-11T17:12:21.5715341Z   info : An error occurred while sending the request.
2017-10-11T17:12:21.5715341Z   info :   The connection with the server was terminated abnormally
2017-10-11T17:12:21.7754943Z   info :   PUT https://dotnet.myget.org/F/dotnet-core/symbols/api/v2/package/
2017-10-11T17:12:22.5484459Z   info :   Conflict https://dotnet.myget.org/F/dotnet-core/symbols/api/v2/package/ 772ms
2017-10-11T17:12:22.5796883Z ##[error]EXEC(0,0): Error : Response status code does not indicate success: 409 (The package could not be uploaded. The package "runtime.win-arm.Microsoft.NETCore.DotNetHostResolver 2.1.0-preview1-25811-01" already exists. Overwriting existing packages is forbidden according to the package retention settings for this feed.).
2017-10-11T17:12:22.5796883Z EXEC : error : Response status code does not indicate success: 409 (The package could not be uploaded. The package "runtime.win-arm.Microsoft.NETCore.DotNetHostResolver 2.1.0-preview1-25811-01" already exists. Overwriting existing packages is forbidden according to the package retention settings for this feed.). [E:\A\_work\1524\s\core-setup\publish\publish.proj]
```

**Comment by weshaggard** [comment]
@eerhardt is this a PR you plan to finish? We do need to get all these changes into master soon so if you cannot we should find someone that can.

**Comment by eerhardt** [comment]
Yes, I can finish this PR with the current set of changes.  I am not planning on pulling any new changes from dev/release/2.0.0 in this PR, so someone else will have to do that later.  Probably when we merge to release/2.0.0 as well.

I had been waiting for our `master` official build to stabilize, which it appears it still hasn't.  https://devdiv.visualstudio.com/DevDiv/_build/index?definitionId=3160&_a=completed

The past 5 builds have failed.

**Comment by weshaggard** [comment]
@eerhardt sounds good. I push to get the core-setup build fixed earlier hopefully it got fixed.

**Comment by eerhardt** [comment]
@dotnet-bot test ci please

**Comment by eerhardt** [comment]
Merging.  OSX has SUCCESS on the link, but is stuck archiving.  Linux arm and Tizen legs are passing in the GenPRTest legs with the new .groovy file.

**Pull request merged**